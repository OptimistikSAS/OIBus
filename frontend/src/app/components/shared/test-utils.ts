import { Provider } from '@angular/core';
import { NgbConfig } from '@ng-bootstrap/ng-bootstrap';

const NO_ANIMATION_NGB_CONFIG: NgbConfig = { animation: false };

export const noAnimation: Provider = { provide: NgbConfig, useValue: NO_ANIMATION_NGB_CONFIG };
