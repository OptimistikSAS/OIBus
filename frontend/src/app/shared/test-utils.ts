import { Provider } from '@angular/core';
import { NgbConfig } from '@ng-bootstrap/ng-bootstrap';
import { Page } from './types';

const NO_ANIMATION_NGB_CONFIG: NgbConfig = { animation: false };

export const noAnimation: Provider = { provide: NgbConfig, useValue: NO_ANIMATION_NGB_CONFIG };

export function toPage<T>(content: Array<T>, totalElements: number = content.length, num = 0, size = Math.max(20, totalElements)): Page<T> {
  return {
    content,
    totalElements,
    number: num,
    size,
    totalPages: Math.ceil(totalElements / size)
  };
}

export function emptyPage<T>(size = 20): Page<T> {
  return {
    totalElements: 0,
    content: [] as Array<T>,
    size,
    totalPages: 0,
    number: 0
  };
}
