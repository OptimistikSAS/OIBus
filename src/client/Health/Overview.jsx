import React from 'react'
import { Row, Col, Container } from 'reactstrap'
import { Link } from 'react-router-dom'
import { ConfigContext } from '../context/configContext.jsx'
import PointsButton from '../South/PointsButton.jsx'

const Overview = () => {
  const { activeConfig } = React.useContext(ConfigContext)
  const applications = activeConfig?.north?.applications
  const dataSources = activeConfig?.south?.dataSources
  return (
    <Container>
      <Row>
        {applications?.map((application) => (
          <Col key={application.applicationId} className={`tight text-${application.enabled ? 'success' : 'muted'}`}>
            <div className="oi-box d-flex align-items-center">
              <Link to={`/north/${application.applicationId}`}>
                <div>{application.applicationId}</div>
                <div>{`(${application.api})`}</div>
              </Link>
            </div>
          </Col>
        ))}
      </Row>
      <Row>
        <Col className="tight">
          <Link to="/engine">
            <div className="oi-box text-success d-flex align-items-center">Engine</div>
          </Link>
        </Col>
      </Row>
      <Row>
        {dataSources?.map((dataSource) => (
          <Col key={dataSource.dataSourceId} className={`tight text-${dataSource.enabled ? 'success' : 'muted'}`}>
            <div className="oi-box">
              <Link to={`/south/${dataSource.dataSourceId}`}>
                <div>{dataSource.dataSourceId}</div>
                <div>{`(${dataSource.protocol})`}</div>
              </Link>
              <PointsButton dataSource={dataSource} />
            </div>
          </Col>
        ))}
      </Row>
    </Container>
  )
}

export default Overview
