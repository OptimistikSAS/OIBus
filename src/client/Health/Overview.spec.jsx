import React from 'react'
import ReactDOM from 'react-dom'
import { act } from 'react-dom/test-utils'

// need BrowserRouter so Link component is not complaining
import { BrowserRouter } from 'react-router-dom'

import Overview from './Overview.jsx'

import activeConfig from '../../../tests/testConfig'

let container
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  document.body.removeChild(container)
  container = null
})

React.useContext = jest.fn().mockReturnValue({ activeConfig })

describe('Overview', () => {
  test('display overview based on config', () => {
    act(() => {
      ReactDOM.render(<BrowserRouter><Overview /></BrowserRouter>, container)
    })
    expect(container).toMatchSnapshot()
  })
})
