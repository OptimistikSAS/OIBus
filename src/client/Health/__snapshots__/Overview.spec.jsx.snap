// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview display overview based on config 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="tight text-success col"
      >
        <div
          class="oi-box d-flex align-items-center"
        >
          <a
            href="/north/c"
          >
            <div>
              c
            </div>
            <div>
              (Console)
            </div>
          </a>
        </div>
      </div>
      <div
        class="tight text-muted col"
      >
        <div
          class="oi-box d-flex align-items-center"
        >
          <a
            href="/north/monoiconnect"
          >
            <div>
              monoiconnect
            </div>
            <div>
              (OIConnect)
            </div>
          </a>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="tight col"
      >
        <a
          href="/engine"
        >
          <div
            class="oi-box text-success d-flex align-items-center"
          >
            Engine
          </div>
        </a>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="tight text-muted col"
      >
        <div
          class="oi-box"
        >
          <a
            href="/south/CSVServer"
          >
            <div>
              CSVServer
            </div>
            <div>
              (CSV)
            </div>
          </a>
          <button
            class="inline-button autosize oi-points-button btn btn-outline-secondary btn-sm"
            type="button"
          >
            Points (2)
          </button>
        </div>
      </div>
      <div
        class="tight text-muted col"
      >
        <div
          class="oi-box"
        >
          <a
            href="/south/MQTTServer"
          >
            <div>
              MQTTServer
            </div>
            <div>
              (MQTT)
            </div>
          </a>
          <button
            class="inline-button autosize oi-points-button btn btn-outline-secondary btn-sm"
            type="button"
          >
            Points (2)
          </button>
        </div>
      </div>
      <div
        class="tight text-muted col"
      >
        <div
          class="oi-box"
        >
          <a
            href="/south/SimulationServer"
          >
            <div>
              SimulationServer
            </div>
            <div>
              (OPCUA)
            </div>
          </a>
          <button
            class="inline-button autosize oi-points-button btn btn-outline-secondary btn-sm"
            type="button"
          >
            Points (2)
          </button>
        </div>
      </div>
      <div
        class="tight text-muted col"
      >
        <div
          class="oi-box"
        >
          <a
            href="/south/SimulationServerBis"
          >
            <div>
              SimulationServerBis
            </div>
            <div>
              (OPCUA)
            </div>
          </a>
          <button
            class="inline-button autosize oi-points-button btn btn-outline-secondary btn-sm"
            type="button"
          >
            Points (1)
          </button>
        </div>
      </div>
      <div
        class="tight text-muted col"
      >
        <div
          class="oi-box"
        >
          <a
            href="/south/PLC-35"
          >
            <div>
              PLC-35
            </div>
            <div>
              (Modbus)
            </div>
          </a>
          <button
            class="inline-button autosize oi-points-button btn btn-outline-secondary btn-sm"
            type="button"
          >
            Points (5)
          </button>
        </div>
      </div>
      <div
        class="tight text-muted col"
      >
        <div
          class="oi-box"
        >
          <a
            href="/south/PLC-42"
          >
            <div>
              PLC-42
            </div>
            <div>
              (Modbus)
            </div>
          </a>
          <button
            class="inline-button autosize oi-points-button btn btn-outline-secondary btn-sm"
            type="button"
          >
            Points (2)
          </button>
        </div>
      </div>
      <div
        class="tight text-success col"
      >
        <div
          class="oi-box"
        >
          <a
            href="/south/FolderScanner"
          >
            <div>
              FolderScanner
            </div>
            <div>
              (FolderScanner)
            </div>
          </a>
          <button
            class="inline-button autosize oi-points-button btn btn-secondary btn-sm disabled"
            disabled=""
          >
            File
          </button>
        </div>
      </div>
      <div
        class="tight text-muted col"
      >
        <div
          class="oi-box"
        >
          <a
            href="/south/SQLDbToFile"
          >
            <div>
              SQLDbToFile
            </div>
            <div>
              (SQLDbToFile)
            </div>
          </a>
          <button
            class="inline-button autosize oi-points-button btn btn-secondary btn-sm disabled"
            disabled=""
          >
            File
          </button>
        </div>
      </div>
      <div
        class="tight text-muted col"
      >
        <div
          class="oi-box"
        >
          <a
            href="/south/OPC-HDA"
          >
            <div>
              OPC-HDA
            </div>
            <div>
              (OPCHDA)
            </div>
          </a>
          <button
            class="inline-button autosize oi-points-button btn btn-outline-secondary btn-sm"
            type="button"
          >
            Points (3)
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
